test/cqlpy/test_invalid_ann_queries.py

```diff
@@ -60,3 +60,11 @@ def test_ann_query_with_null_vector(cql, test_keyspace):
 
         with pytest.raises(InvalidRequest, match="Unsupported null value for column v"):
             cql.execute(f"SELECT * FROM {table} ORDER BY v ANN OF null LIMIT 5")
+
+@pytest.mark.xfail(reason="We do not support primary key filtering yet")
+def test_ann_query_with_pk_restriction_fails(cql, test_keyspace):
```

Comment:
Something seems wrong here... You call the test "..._fails", and yet you don't check below that anything fails. Even worse, you don't seem to check anything, you just execute some SELECT but don't check what it returns, or if it threw an exception.

Does this test pass in Cassandra? In Scylla? What's going on here?

Reply:
I'll try to make a more concrete suggestion. If you want this test to stay because today it is your only way to check how **Cassandra** works for filtering, then I support this idea, but in that case, I think:

1. You should expand the test to check how Cassandra **really** behaves on these filtering searches. Do they work? Do they work only on key lookups or any column? Or what? It's not interesting (in my opinion) just to check that Cassandra doesn't return an error in this case.
2. You should add a comment in front of this test explaining the situation of this test- that this test can never pass on Scylla (because nothing in the cqlpy framework runs the vector search backend) so it's only useful as a way to understand how Cassandra behaves in this case.